File: src/main/java/mods/eln/sim/electrical/nbt/NbtResistor.java
package mods.eln.sim.electrical.nbt;

import mods.eln.misc.INBTTReady;
import mods.eln.sim.electrical.mna.component.Resistor;
import mods.eln.sim.electrical.mna.state.State;
import net.minecraft.nbt.NBTTagCompound;

public class NbtResistor extends Resistor implements INBTTReady {

    String name;

    public NbtResistor(String name, State aPin, State bPin) {
        super(aPin, bPin);
        this.name = name;
    }

    @Override
    public void readFromNBT(NBTTagCompound nbt, String str) {
        name += str;
        setR(nbt.getDouble(str + "R"));
    }

    @Override
    public void writeToNBT(NBTTagCompound nbt, String str) {
        name += str;
        nbt.setDouble(str + "R", getR());
    }
}

File: src/main/java/mods/eln/sim/nbt/NbtResistor.kt
package mods.eln.sim.nbt

import mods.eln.misc.INBTTReady
import mods.eln.sim.mna.component.Resistor
import mods.eln.sim.mna.state.State
import net.minecraft.nbt.NBTTagCompound

class NbtResistor(name: String, aPin: State?, bPin: State?) : Resistor(aPin, bPin), INBTTReady {

    init {
        this.name = name
    }

    override fun readFromNBT(nbt: NBTTagCompound, str: String) {
        name += str
        r = nbt.getDouble(str + "R")
    }

    override fun writeToNBT(nbt: NBTTagCompound, str: String) {
        name += str
        nbt.setDouble(str + "R", r)
    }
}
